bin/generate_tool_stats.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python
+
+import argparse
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from PIL import Image
+from wordcloud import WordCloud
+
+
+def get_last_url_position(toot_id: str) -> str:
+    """
+    Returns the last url position of the toot_id, if the value is not a
+    url it returns the toot_id. So works for local and toolshed
+    installed tools.
+
+    :param tool_id: galaxy tool id
+    """
+
+    toot_id = toot_id.split("/")[-1]
+    return toot_id
+
+
+def add_tool_stats_to_tools(tool_stats_path: str, tools_path: str, output_path: str, column_name: str) -> pd.DataFrame:
```

Comment:
What about adding that function directly in the other script? Then we have everything in the `all_tools.tsv` file
And then this script focuses on the wordcloud plotting. What do you think?